e2e/cypress/integration/bots/managing_bot_accounts_spec.js

```diff
@@ -0,0 +1,311 @@
+
+    it('MM-T1859 Bot is kept active when owner is disabled', () => {
+        // # Visit bot config
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/integrations/bot_accounts');
+
+        // # Click 'false' to disable
+        cy.findByTestId('ServiceSettings.DisableBotsWhenOwnerIsDeactivatedfalse').click();
+
+        // # Save
+        cy.findByTestId('saveSetting').should('be.enabled').click();
+
+        // # Create another admin account
+        const adminName = 'other-admin-' + Date.now();
+        let newAdminId;
+        cy.apiCreateCustomAdmin(adminName).then((result) => {
+            if (!result.sysadmin.email_verified) {
+                newAdminId = result.sysadmin.id;
+                return cy.apiVerifyUserEmailById(result.sysadmin.id);
+            }
+            return Promise.resolve();
+        }).then(() => {
+            // # Login as the new admin
+            cy.apiAdminCustomLogin(adminName);
+
+            // # Create a new bot as the new admin
+            const botName3 = 'stay-enabled-bot-' + Date.now();
+            cy.apiCreateBot(botName3, 'Bot That Stays Enabled', 'hello bot');
+
+            // # Login again as main admin
+            cy.apiAdminLogin();
+
+            // # Deactivate the newly created admin
+            cy.apiDeactivateUser(newAdminId);
+
+            // # Get bot list
+            cy.visit(`/${newTeam.name}/integrations/bots`);
+
+            // # Search for the other bot
+            cy.get('#searchInput').type('Bot That Stays Enabled');
+
+            // * Validate that the plugin is still active, even though it's owner is disabled
+            cy.get('.bot-list__disabled').should('not.be.visible');
+            cy.findByText(`Bot That Stays Enabled (@${botName3})`).scrollIntoView().should('be.visible');
+
+            cy.visit(`/${newTeam.name}/messages/@sysadmin`);
+
+            // # Get last post message text
+            cy.getLastPostId().then((postId) => {
+                cy.get(`#postMessageText_${postId}`).as('postMessageText');
+            });
+
+            // * Verify entire message
+            cy.get('@postMessageText').
+                should('be.visible').
+                and('contain.text', `${adminName} was deactivated. They managed the following bot accounts`).
+                and('contain.text', botName3);
+        });
+    });
+
+    it('MM-T1860 Bot is disabled when owner is deactivated', () => {
+        cy.apiAdminLogin();
+
+        // # Create another admin account
+        const adminName = 'other-admin-' + Date.now();
+        let newAdminId;
+        cy.apiCreateCustomAdmin(adminName).then((result) => {
+            if (!result.sysadmin.email_verified) {
+                newAdminId = result.sysadmin.id;
+                return cy.apiVerifyUserEmailById(result.sysadmin.id);
+            }
+            return Promise.resolve();
+        }).then(() => {
+            // # Login as the new admin
+            cy.apiAdminCustomLogin(adminName);
```

Comment:
Please see comment on `apiCreateCustomAdmin`. Then this could simply,
```javascript
        cy.apiCreateCustomAdmin().then(({sysadmin}) => {
            // # Login as the new admin
            cy.apiLogin(sysadmin);
            ...
            cy.apiDeactivateUser(sysadmin.id);
```

Ypu might not need `cy.apiVerifyUserEmailById` unless you're changing config to require email verification.